### PR TITLE
cd: add GoCD deployment pipeline

### PR DESCRIPTION
**Pull request by joshuarli**
This adds deployment configurations for GoCD, the new internal continuous deployment tool. Please refer to the [Freight Migration Guide](https://www.notion.so/sentry/Freight-Migration-Guide-2bd47bb683254077ad31e7ad625345ae) for info on how to coordinate GoCD deploys during this transitional phase.

#skip-changelog

### CODE REVIEW

File: gocd/pipelines/symbolicator.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                      checks:
+                          timeout: 1800
+                          elastic_profile_id: symbolicator
+                          tasks:
+                              - script: |
+                                    /devinfra/scripts/checks/githubactions/checkruns.py \
+                                    getsentry/symbolicator \
+                                    ${GO_REVISION_SYMBOLICATOR_REPO} \
+                                    'Unit Tests' \
+                                    'Integration Tests'
```
Comment (Swatinem) [inline gocd/pipelines/symbolicator.yaml:38]: This should be `Tests` and `Sentry-Symbolicator Tests` by now

Comment (joshuarli) [reply gocd/pipelines/symbolicator.yaml:38]: Ah, good catch!

**Comment by joshuarli** [comment]
I'm noticing there's a cloudbuild trigger `gcb-docker-image` building and uploading to `us.gcr.io/sentryio/symbolicator` which is currently being used in prod.

I can't find anything using `ghcr.io/getsentry/symbolicator`? Just curious.

**Comment by Swatinem** [comment]
I have no idea tbh, I’m not involved in the migration from gcr to ghcr. (I just heart that it might be a thing?)

@asottile-sentry can probably shed some light there?

All in all, I know next to nothing about our current deployment infrastructure. Maybe @oioki or @beezz can help out with reviews here.

**Comment by asottile-sentry** [comment]
for prod we still want `gcr` so it says within gcp, for dev / open source we're trying to standardize on ghcr so we can lock down gcr further

**Review by ashwoods — APPROVED** [review]

**Comment by joshuarli** [comment]
hold off on merging - i've shipped some improvements since which enable pipelines to be refactored a bit

**Pull request merged**